internal/pam_issuer_server/handler.go

```diff
@@ -0,0 +1,389 @@
+//go:build linux
+
+package pam_issuer_server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	pamapi "github.com/flightctl/flightctl/api/v1alpha1/pam-issuer"
+	"github.com/flightctl/flightctl/internal/auth/issuer/pam"
+	"github.com/flightctl/flightctl/internal/config"
+	"github.com/flightctl/flightctl/internal/crypto"
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+)
+
+// Handler implements the PAM issuer API handlers
+type Handler struct {
+	log         logrus.FieldLogger
+	cfg         *config.Config
+	pamProvider *pam.PAMOIDCProvider
+	cancel      context.CancelFunc
+	wg          sync.WaitGroup
+}
+
+// NewHandler creates a new PAM issuer handler
+func NewHandler(
+	log logrus.FieldLogger,
+	cfg *config.Config,
+	ca *crypto.CAClient,
+) (*Handler, error) {
+	if cfg.Auth == nil || cfg.Auth.PAMOIDCIssuer == nil {
+		return nil, fmt.Errorf("PAM OIDC issuer not configured")
+	}
+
+	pamProvider, err := pam.NewPAMOIDCProvider(ca, cfg.Auth.PAMOIDCIssuer)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create PAM OIDC provider: %w", err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	h := &Handler{
+		log:         log,
+		cfg:         cfg,
+		pamProvider: pamProvider,
+		cancel:      cancel,
+	}
+
+	// Start cleanup goroutine
+	h.wg.Add(1)
+	go func() {
+		defer h.wg.Done()
+		ticker := time.NewTicker(5 * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				pamProvider.CleanupExpiredCodes()
+				pamProvider.CleanupExpiredSessions()
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return h, nil
+}
+
+// Close cleans up resources
+func (h *Handler) Close() {
+	if h.cancel != nil {
+		h.cancel()
+	}
+	h.wg.Wait()
+	if h.pamProvider != nil {
+		h.pamProvider.Close()
+	}
+}
+
+// Helper function to write JSON response
+func writeJSON(w http.ResponseWriter, statusCode int, data interface{}) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(statusCode)
+	if err := json.NewEncoder(w).Encode(data); err != nil {
+		logrus.Errorf("Failed to encode JSON response: %v", err)
+	}
+}
+
+// Helper function to write error response
+func writeError(w http.ResponseWriter, statusCode int, message string) {
+	status := pamapi.Status{
+		ApiVersion: "v1alpha1",
+		Kind:       "Status",
+		Code:       int32(statusCode), //nolint:gosec // HTTP status codes are always within int32 range
+		Message:    message,
+		Reason:     "Error",
+		Status:     "Failure",
+	}
+	writeJSON(w, statusCode, status)
+}
+
+// extractSessionContext extracts session information from HTTP request and adds it to context
+func (h *Handler) extractSessionContext(ctx context.Context, r *http.Request) context.Context {
+	// Extract session cookie (primary method for session management)
+	if cookie, err := r.Cookie("session_id"); err == nil && cookie.Value != "" {
```

Comment:
what if err != do we care?

Reply:
no , we handle this downstream , if not there it might just means that the user hasnt logged in yet